index.d.ts

```diff
@@ -0,0 +1,45 @@
+import { TextStyle } from "react-native";
+import React from "react";
+
+/**
+ * Escape characters which are meaningful in regular expressions
+ */
+export type AutoEscape = boolean;
+
+/**
+ * Styles applied to highlighted text
+ */
+export interface HighlightStyle extends TextStyle {}
+
+/**
+ * Process each search word and text to highlight before comparing (eg remove accents);
+ */
+export type Sanitize = (text: string) => string;
+
+/**
+ * Array of search words
+ */
+export type SearchWords = string[];
+
+/**
+ * Styles applied to the text wrapper
+ */
+export interface Style extends TextStyle {}
+
+/**
+ * Text to highlight matches in
+ */
+export type TextToHighlight = string;
+
+export interface HighlighterProps {
+  autoEscape?: AutoEscape;
+  highlightStyle?: HighlightStyle;
```

Comment:
Couldn't we use a shortcut here and less specific types like

```diff
-highlightStyle?: HighlightStyle;
+highlightStyle?: StyleProp<TextStyle>;
```

same for all the other props. When its just a boolean why not specify just a boolean?

Reply:
yeah, it's gorgeous! I wrote a bunch of unnecessary code here. So, I really appreciate your review. All of these props are very very simple, so, doesn't make any sense creating extras `types/interfaces`. Again, thanks.

Reply:
And it goes one better even. Since the control supports all the TextProps we can make this typing really shine:

```ts
  import { TextStyle, StyleProp, TextProps } from "react-native"
  import React from "react"

  type HighlighterProps = {
    /**
     * Escape characters which are meaningful in regular expressions
    */
    autoEscape?: boolean

    /**
     * Styles applied to highlighted text
     */
    highlightStyle?: StyleProp<TextStyle>

    /**
    * Process each search word and text to highlight before comparing (eg remove accents);
    */
    sanitize?: (text: string) => string

    /**
    * Array of search words
    */
    searchWords: string[]

    /**
    * Styles applied to the text wrapper
    */
    style?: StyleProp<TextStyle>

    /**
    * Text to highlight matches in
    */
    textToHighlight: string
  } & TextProps

  const Highlighter: React.FC<HighlighterProps>

  export default Highlighter
}
```

Note the `& TextProps` part.

This allows the typing to accept: `<Highlighter onPress={() => null} .../>` and other `Text` props among its own.

I am using this type definition in my app and it works great.

Reply:
Got it. Thanks a lot for your great help! You improved it a lot here.

Just another help. With the code you wrote I'm getting the message:
```ts
Top-level declarations in .d.ts files must start with either a 'declare' or 'export' modifier.ts(1046)
```

So, I added `declare const Highlighter: React.FC<HighlighterProps>;`, does it make sense?

Reply:
The component only exports the `Highlighter` as default item, therefore `declare` is correct. If it would also export the component as named export `Highlighter` then we would have to use `export`.

I think you can shorten the declaration then to: `export default Highlighter: React.FC<HighlighterProps>`

Does this work?

Reply:
Unfortunately not. Tried to reply, but it doesn't work. Another way, that at least I found could be:

```ts
declare module 'Highlighter' {
  type HighlighterProps = {
...
  } & TextProps;

  const Highlighter: React.FC<HighlighterProps>;
  export default Highlighter
}
```
But the current one works fine.

ref: https://www.typescriptlang.org/docs/handbook/modules.html#default-exports